File: ruby/send_document.rb
#!/usr/bin/env ruby
# إرسال مستند عبر واتساب
# Send document via WhatsApp
# واٹس ایپ کے ذریعے دستاویز بھیجیں

require "net/http"
require "uri"
require "json"
require "base64"

# إعدادات الاتصال - Connection settings - رابطہ کی ترتیبات
APP_KEY = "your_app_key"
API_SECRET = "your_api_secret"
PROJECT_ID = "your_project_id"
RECIPIENT = "9665XXXXXXXX"

# رابط المستند والوصف واسم الملف - Document URL, caption and filename - دستاویز کا لنک، کیپشن اور فائل کا نام
DOCUMENT_URL = "https://example.com/document.pdf"
DOCUMENT_CAPTION = "وصف المستند"
DOCUMENT_FILENAME = "document.pdf"

def send_document_message
  # إرسال مستند - Send document - دستاویز بھیجیں
  url = URI("https://api-users.4jawaly.com/api/v1/whatsapp/#{PROJECT_ID}")
  auth_string = "#{APP_KEY}:#{API_SECRET}"
  auth_b64 = Base64.strict_encode64(auth_string)

  payload = {
    path: "global",
    params: {
      url: "messages",
      method: "post",
      data: {
        messaging_product: "whatsapp",
        to: RECIPIENT,
        type: "document",
        document: {
          link: DOCUMENT_URL,
          caption: DOCUMENT_CAPTION,
          filename: DOCUMENT_FILENAME
        }
      }
    }
  }

  http = Net::HTTP.new(url.host, url.port)
  http.use_ssl = true
  http.verify_mode = OpenSSL::SSL::VERIFY_PEER

  request = Net::HTTP::Post.new(url)
  request["Content-Type"] = "application/json"
  request["Authorization"] = "Basic #{auth_b64}"
  request.body = payload.to_json

  begin
    response = http.request(request)
    puts "رمز الاستجابة / Response code / جوابی کوڈ: #{response.code}"
    puts "\nالاستجابة / Response / جواب:\n#{JSON.pretty_generate(JSON.parse(response.body))}"
  rescue Net::OpenTimeout, Net::ReadTimeout, SocketError => e
    puts "خطأ في الاتصال / Connection error / رابطہ میں خرابی: #{e.message}"
  rescue JSON::ParserError => e
    puts "خطأ في تحليل الاستجابة / Response parse error / جواب کی تشریح میں خرابی: #{e.message}"
    puts "النص الخام / Raw text / خام متن: #{response&.body}"
  end
end

send_document_message if __FILE__ == $PROGRAM_NAME

File: rust/src/send_document.rs
// إرسال مستند عبر واتساب
// Send document via WhatsApp
// واٹس ایپ کے ذریعے دستاویز بھیجیں

use crate::api;
use serde_json::json;

/// إرسال مستند
/// Send document message
/// دستاویز بھیجیں
pub fn send_document(
    config: &api::Config,
    to: &str,
    link: &str,
    caption: Option<&str>,
    filename: Option<&str>,
) -> Result<(), Box<dyn std::error::Error>> {
    let mut document = json!({
        "link": link
    });

    if let Some(cap) = caption {
        document["caption"] = json!(cap);
    }

    if let Some(name) = filename {
        document["filename"] = json!(name);
    }

    let data = json!({
        "messaging_product": "whatsapp",
        "to": to,
        "type": "document",
        "document": document
    });

    api::make_api_call(config, data)
}
